TypeScript: getRepoByName return type missing owner property causes error in analyze repo page

## Problem

`apps/web/lib/server/internal-api.ts` function `getRepoByName` returns an object with `owner_login` and `owner_id` properties, but `apps/web/app/analyze/(repo)/[owner]/[repo]/page.tsx` tries to access `repoInfo.owner?.login`.

**Files:**
- `apps/web/lib/server/internal-api.ts` line 111-127 (return type definition)
- `apps/web/app/analyze/(repo)/[owner]/[repo]/page.tsx` line 53 (usage)

**Error:**
```
app/analyze/(repo)/[owner]/[repo]/page.tsx(53,26): error TS2339: Property 'owner' does not exist on type '{ id: number; full_name: string; description: string; language: string; license: string; forks: number; stars: number; owner_login: string; owner_id: number; default_branch: string; }'.
```

**Current return type:**
```typescript
{
  id: number;
  full_name: string;
  description: string;
  language: string;
  license: string;
  forks: number;
  stars: number;
  owner_login: string;
  owner_id: number;
  default_branch: string;
}
```

**Usage causing error:**
```typescript
type: repoInfo.owner?.login ? 'Organization' : 'Person',  // Error: Property 'owner' does not exist
```

## Impact

- TypeScript compilation fails with error TS2339
- The code logic is trying to determine if the repo owner is an organization or person, but the property name mismatch causes a type error
- May cause runtime issues if the code is forced through with type assertions

## Suggested Fix

Option 1: Add `owner` property to the return type (recommended):

```typescript
{
  id: number;
  full_name: string;
  description: string;
  language: string;
  license: string;
  forks: number;
  stars: number;
  owner_login: string;
  owner_id: number;
  default_branch: string;
  owner?: { login: string };  // Add this property
}
```

Option 2: Update the usage to use `owner_login` directly:

```typescript
type: repoInfo.owner_login ? 'Organization' : 'Person',
```

--- a/apps/web/components/Analyze/context.ts
+++ b/apps/web/components/Analyze/context.ts
@@ -12,6 +12,7 @@ export interface RepoInfo {
   license: string;
   forks: number;
   stars: number;
+  default_branch: string;
   owner: {
     login: string;
     avatar_url: string;

--- a/apps/web/lib/server/internal-api.ts
+++ b/apps/web/lib/server/internal-api.ts
@@ -59,7 +59,24 @@ export function jsonResponse(body: unknown, init?: ResponseInit) {
   });
 }
 
-export async function getRepoById(id: number | string, signal?: AbortSignal) {
+export interface RepoDetail {
+  id: number;
+  full_name: string;
+  description: string;
+  language: string;
+  license: string;
+  forks: number;
+  stars: number;
+  owner_login: string;
+  owner_id: number;
+  default_branch: string;
+  owner: {
+    login: string;
+    avatar_url: string;
+  };
+}
+
+export async function getRepoById(id: number | string, signal?: AbortSignal): Promise<RepoDetail | null> {
   const row = await executeOneRow<{
     id: number;
     full_name: string;
@@ -106,7 +123,7 @@ export async function getRepoById(id: number | string, signal?: AbortSignal) {
   };
 }
 
-export async function getRepoByName(owner: string, repo: string, signal?: AbortSignal) {
+export async function getRepoByName(owner: string, repo: string, signal?: AbortSignal): Promise<RepoDetail | null> {
   const row = await executeOneRow<{
     id: number;
     full_name: string;

